platform/api/oci/types/volume_test.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateVolumeInput(t *testing.T) {
+	type in struct {
+		input CreateVolumeInput
+	}
+	type out struct {
+		jsonStr string
+	}
+
+	tests := []struct {
+		in   in
+		out  out
+		name string
+	}{
+		{
+			in: in{input: CreateVolumeInput{
+				AvailabilityDomain: "Uocm:PHX-AD-1",
+				CompartmentID:      "ocid1.compartment.oc1..aaaaaaaayzfqeibduyox6iib3olcmdar3ugly4fmameq4h7lcdlihrvur7xq",
+				DisplayName:        StrToPtr("MyCustomVolume"),
+				SizeInMBs:          IntToPtr(2048),
+				VolumeBackupID:     StrToPtr("ocid1.volume.oc1..fkldasjflkasjdflkasjfldsajflajsdlfjasldfjasldfjasdlfjaasdfds"),
+			}},
+			out:  out{jsonStr: `{"availabilityDomain":"Uocm:PHX-AD-1","compartmentId":"ocid1.compartment.oc1..aaaaaaaayzfqeibduyox6iib3olcmdar3ugly4fmameq4h7lcdlihrvur7xq","displayName":"MyCustomVolume","sizeInMBs":2048,"volumeBackupId":"ocid1.volume.oc1..fkldasjflkasjdflkasjfldsajflajsdlfjasldfjasldfjasdlfjaasdfds"}`},
+			name: "All Fields",
+		},
+		{
+			in: in{input: CreateVolumeInput{
+				AvailabilityDomain: "Uocm:PHX-AD-1",
+				CompartmentID:      "ocid1.compartment.oc1..aaaaaaaayzfqeibduyox6iib3olcmdar3ugly4fmameq4h7lcdlihrvur7xq",
+			}},
+			out:  out{jsonStr: `{"availabilityDomain":"Uocm:PHX-AD-1","compartmentId":"ocid1.compartment.oc1..aaaaaaaayzfqeibduyox6iib3olcmdar3ugly4fmameq4h7lcdlihrvur7xq"}`},
+			name: "Required Only",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			marshal, err := json.Marshal(test.in.input)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if string(marshal) != test.out.jsonStr {
```

Comment:
If we're using testify, this should be `asesrt.Equal`

Reply:
:+1: